### PR TITLE
Add dimension support for GetPrint

### PR DESCRIPTION
**Pull request by m-kuhn**
Adds support for dimension to GetPrint requests.

### CODE REVIEW

File: src/server/qgsdimensionfilter.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#include "qgsdimensionfilter.h"
```
Comment (m-kuhn) [inline src/server/qgsdimensionfilter.cpp:1]: TODO Headers

File: src/server/qgsdimensionfilter.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#ifndef QGSDIMENSIONFILTER_H
+#define QGSDIMENSIONFILTER_H
+
+#include "qgsfeaturefilterprovider.h"
+#include "qgis_server.h"
+
+class SERVER_EXPORT QgsDimensionFilter : public QgsFeatureFilterProvider
```
Comment (m-kuhn) [inline src/server/qgsdimensionfilter.h:7]: TODO docs

**Comment by nirvn** [comment]
@m-kuhn , there's a reference image missing in there.

**Comment by m-kuhn** [comment]
Yep, thanks. Will take care of that!

It's still very drafty, just online to give others the possibility to show excitement or give a no-go.

**Comment by m-kuhn** [comment]
@elpaso @rldhont how does that look to you?

**Comment by nyalldawson** [comment]
Is this a feature or fix?

**Comment by m-kuhn** [comment]
I'd treat this as a fix. Dimension enabled projects with GetPrint are currently unusable, just rendering everything on top of each other.

**Review by elpaso — APPROVED** [review]

**Pull request merged**